astropy/nddata/nduncertainty.py

```diff
@@ -186,6 +186,18 @@ def __repr__(self):
             body = str(self.array)
         return ''.join([prefix, body, ')'])
 
+    def __reduce__(self):
+        """Because of the weak reference the class wouldn't be picklable."""
+        try:
+            return (type(self),
+                    (self._array, False, self._unit),
+                    (self.parent_nddata, ))
+        except MissingDataAssociationException:
+            return type(self), (self._array, False, self._unit)
```

Comment:
Any reason to use `__reduce__` instead of `__getstate__`, which would probably be more clear ? 
And I guess that `MissingDataAssociationException` is triggered when `parent_nddata` is missing, but a comment could help here.

Reply:
I think I tried `__getstate__` and I had some problems. But I can test it again this evening.

Reply:
Yes, the exception is raised if there's no parent. I could add a comment (and test) for that.

Reply:
Seems like it also works with `__getstate__`, no idea why I changed it.